src/NuGet.Core/NuGet.Frameworks/comparers/NuGetFrameworkFullComparer.cs

```diff
@@ -32,7 +32,8 @@ public bool Equals(NuGetFramework x, NuGetFramework y)
 
             return x.Version == y.Version
                    && StringComparer.OrdinalIgnoreCase.Equals(x.Framework, y.Framework)
-                   && StringComparer.OrdinalIgnoreCase.Equals(x.Profile, y.Profile);
+                   && StringComparer.OrdinalIgnoreCase.Equals(x.Profile, y.Profile)
+                   && !x.IsUnsupported;
```

Comment:
This feels weird. 
This is an equals method, but this statement only looks at the left side. 

Additionally IsUnsupported seems like a helper, rather than something part of the NuGetFramework. IT's not a field, so as such it shouldn't be a part of this comparison.

Reply:
if it got to the line with unsupported, everything is equals, so i could check the left side or right side for unsupported at that point.

could make this use IsInvalid instead, but i thought the consolidation with IsUnsupported was what the people wanted.

Reply:
> could make this use IsInvalid instead, but i thought the consolidation with IsUnsupported was what the people wanted.

My thought process is, currently bad framework parse to unsupported, if we keep that, then we don't need to update methods like Equals. Minimzes the risk of regressions.

Reply:
@nkolev92 and i chatted over teams. his basic feedback was to wait on IsInvalid or something better than IsUnsupported for some other time... (NuGet/Home#5413 covers that existing problem)

fixed in next commit

Reply:
Has the latest commit been pushed?
Specifically:
```cs
                   && !x.IsUnsupported;
``` 
is not necessary